### PR TITLE
management startup + sidebar popluation

### PR DESCRIPTION
**Pull request by manthankumbhar**
### In this PR:
1. Have setup the sidebar population using a dummy array.
2. Designed and added CSS for the same.
3. Added some functionalities around options clicking (basic redirecting + a button to create new documents), will set up properly when API is done.
Attaching a picture to show how the sidebar looks below.
Screenshot from 2022-04-16 03-45-27

### CODE REVIEW

File: src/Components/Sidebar/Sidebar.js [hunk 1]
```diff
@@ -1,4 +1,4 @@
-import React, { useCallback } from "react";
+import React, { useCallback, useState } from "react";
 import "Components/Sidebar/Sidebar.scss";
```
Comment (kavadithya) [inline src/Components/Sidebar/Sidebar.js]: minor: why is components folder with a capital C? It's bothering me every time I see this repo's code

File: src/Components/Sidebar/Sidebar.js [hunk 2]
```diff
@@ -14,12 +14,44 @@ export default function Sidebar() {
...
+        key={key}
+        onClick={() => console.log(`backend/document/${item.id} is clicked`)}
+        // will convert it to another function and add useCallback
+      >
+        {item.name}
+      </div>
+    );
+  });
+
+  const SidebarMenu = useCallback(() => {
```
Comment (kavadithya) [inline src/Components/Sidebar/Sidebar.js]: why captial S?

Comment (kavadithya) [reply src/Components/Sidebar/Sidebar.js]: also functions are better named with the action they do; like openSideBar, closeSidebar, killManthan etc.

File: src/Components/Sidebar/Sidebar.js [hunk 3]
```diff
@@ -14,12 +14,44 @@ export default function Sidebar() {
...
       <Link to="/" className="sidebar__logo">
         <img src={logo} alt="logo" />
       </Link>
+      <div className="sidebar__menu">
+        <div className="sidebar__menu--headers">
+          <p className="sidebar__menu--headers--title">Documents:</p>
+          <button className="sidebar__menu--headers--btn">+</button>
+          {/* will set onClick functionalities once api is done */}
+        </div>
+        <SidebarMenu />
```
Comment (kavadithya) [inline src/Components/Sidebar/Sidebar.js]: SideBarMenu is a function - should not be used as a component; can cause unspecified behaviour;

Comment (kavadithya) [reply src/Components/Sidebar/Sidebar.js]: if it's supposed to be a component, then don't use useCallback; 
useCallback is for optimizing functions - I'm not clear on how it affects components - if you found any documentation on that, I'd like to see

**Review by kavadithya — COMMENTED** [review]
comments

File: src/Components/Sidebar/Sidebar.js (hunk 2 above)
Comment (manthankumbhar) [reply src/Components/Sidebar/Sidebar.js]: okaysir

File: src/Components/Sidebar/Sidebar.js (hunk 3 above)
Comment (manthankumbhar) [reply src/Components/Sidebar/Sidebar.js]: No, it does on functions only, components inside are optimized.

**Review by kavadithya — APPROVED** [review]

**Pull request closed without merging**